### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by shosuke-13**
### **PR Type**
enhancement, bug fix

___

### **Description**
- Reordered the list of models in several scripts for consistency.
- Changed the fine-tuning method from `ia3` to `lora` across multiple scripts.
- Adjusted the learning rate from `3e-3` to `1e-4` for improved training performance.
- Enhanced the `train.py` script to include better error handling when saving results to S3.
- Updated the logic for generating unique run names to include the fine-tuning method.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>exp2_gene.sh</strong><dd><code>Update model order and fine-tuning parameters in gene script</code></dd></summary>
<hr>

finetune/scripts/dok/exp2_gene.sh

<li>Reordered model list for consistency.<br> <li> Changed fine-tuning method from <code>ia3</code> to <code>lora</code>.<br> <li> Adjusted learning rate from <code>3e-3</code> to <code>1e-4</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-c2529b83453a1b59ef4710c7c8a2bcac5484881b394e277ce774d1d210d01a2e">+5/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_lncrna.sh</strong><dd><code>Update model order and fine-tuning parameters in lncrna script</code></dd></summary>
<hr>

finetune/scripts/dok/exp2_lncrna.sh

<li>Reordered model list for consistency.<br> <li> Changed fine-tuning method from <code>ia3</code> to <code>lora</code>.<br> <li> Adjusted learning rate from <code>3e-3</code> to <code>1e-4</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-c240a01021bd80f0495797ada58292c790a9cafc20bd39d705032f43b960230c">+5/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_splicing.sh</strong><dd><code>Update model order and fine-tuning parameters in splicing script</code></dd></summary>
<hr>

finetune/scripts/dok/exp2_splicing.sh

<li>Reordered model list for consistency.<br> <li> Changed fine-tuning method from <code>ia3</code> to <code>lora</code>.<br> <li> Adjusted learning rate from <code>3e-3</code> to <code>1e-4</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-d96b1e011c3aa1986b78bd19f90d62f0f30f1f9df30dee3a0a4e3ab28332c873">+5/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_tasks.sh</strong><dd><code>Update model order and fine-tuning parameters in tasks script</code></dd></summary>
<hr>

finetune/scripts/dok/exp2_tasks.sh

<li>Reordered model list for consistency.<br> <li> Changed fine-tuning method from <code>ia3</code> to <code>lora</code>.<br> <li> Adjusted learning rate from <code>3e-3</code> to <code>1e-4</code>.<br> <li> Modified logging steps from <code>10000</code> to <code>15000</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-91a001f3c05a5d11ce7eaa70d58a4b808784a40e4001a69ead955d9e8efc0cad">+8/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>train.py</strong><dd><code>Enhance S3 saving logic and error handling in training script</code></dd></summary>
<hr>

finetune/train.py

<li>Added <code>bucket_name</code> parameter to <code>save_results</code>.<br> <li> Improved error handling for S3 operations.<br> <li> Enhanced run name generation with <code>use_peft</code> parameter.<br> <li> Adjusted logic for saving predictions and metrics to S3.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016">+49/-34</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by shosuke-13** [comment]
Preparing PR description...

**Comment by shosuke-13** [comment]
Preparing review...

**Comment by shosuke-13** [comment]
## PR Reviewer Guide 🔍

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 4 🔵🔵🔵🔵⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Key issues to review</strong><br><br>

<a href='https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016R255-R315'><strong>Possible Bug</strong></a><br>The `save_results` function has a new parameter `predictions` which is not clearly defined in the context of its usage. Ensure that the type and structure of `predictions` are well understood and documented.

<a href='https://github.com/shosuke-13/kmer-tokenization-strategy/pull/6/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016R304-R306'><strong>Code Smell</strong></a><br>The error handling in the `save_results` function could be improved. Instead of exiting the program on an error, consider raising an exception or returning an error status to allow for better error management.

</td></tr>
</table>

**Comment by shosuke-13** [comment]
## Generating PR code suggestions


Work in progress ...<br>
